File: RxJavaDemo/app/src/main/java/com/ucar/sycao/eventbus/SecondActivity.java
package com.ucar.sycao.eventbus;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.ucar.sycao.R;

import org.greenrobot.eventbus.EventBus;

public class SecondActivity extends AppCompatActivity {

    Button btnSecond;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        btnSecond=(Button)findViewById(R.id.btn_second);
        btnSecond.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                EventBus.getDefault().post(new FirstEvent("这是从第二个Activity发送来的消息"));
                Toast.makeText(getApplicationContext(),"send msg",Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: KotlinDemo/app/src/main/java/com/corey/demo/SecondActivity.kt
package com.corey.demo

import android.app.Activity
import android.os.Bundle
import android.util.Log
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.Disposable
import io.reactivex.observers.DisposableObserver
import io.reactivex.schedulers.Schedulers
import org.jetbrains.anko.*
import org.jetbrains.anko.sdk25.coroutines.onClick
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

/**
 * Created by sycao on 2017/8/24.
 *
 */
class SecondActivity : Activity() {

    lateinit var disposable: Disposable

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val layout = verticalLayout {
            val title = textView()
            button("this is a button") {
                onClick {
                    title.text = "test"
                    getData()
                }
            }.lparams {
                width = matchParent
                verticalMargin = dip(5)
            }
        }

        setContentView(layout)
    }

    private fun getData() {
        val baseUrl = "https://api.douban.com/v2/"
        val retrofit = Retrofit.Builder()
                .baseUrl(baseUrl)
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .build()
        val movieService = retrofit.create(MovieService::class.java)

        disposable = movieService.getMovieSubjectById("1292052")
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribeWith(object : DisposableObserver<MovieEntity.SubjectsBean>() {
                    override fun onError(e: Throwable) {
                        Log.d("rxKotlin", "onError")
                    }

                    override fun onComplete() {
                        Log.d("rxKotlin", "onComplete")
                    }

                    override fun onNext(t: MovieEntity.SubjectsBean) {
                        toast(applicationContext, t.id)
                    }
                })
    }

    override fun onDestroy() {
        super.onDestroy()
        disposable.dispose()
    }
}